graphql-hive.gemspec

```
@@ -24,7 +24,7 @@ Gem::Specification.new do |spec|
     `git ls-files -z`.split("\x0").reject { |f| f.match(%r{\A(?:test|spec|features)/}) }
   end
 
-  spec.add_dependency 'graphql', '~> 2.0.9'
+  spec.add_dependency 'graphql', '< 3',  '>= 2'
```

```suggestion
  spec.add_dependency 'graphql', '~> 2'
```

(nit, just more idiomatic)

---

I had this before, and @cassidycodes and I thought that this would be more clear! I don't have a particular preference but yep, they both mean the same thing.

---

@charcoalyy and I were debugging an issue with the usage reporter and printer yesterday. 

So I had to come back to this one. I had to make it `~> 2.3` (which is `2.3.x`) because the `GraphQL::Printer` works very differently in 2.0 vs 2.3, and we need it to be consistent in all services that use Hive. It is unfortunately a breaking change between 2.0 to 2.3 as far as the Hive's usage of the GraphQL Printer is concerned. 

This is the best solution I can think of right now, but potentially we can make it `>= 2.3` as well.

The usage_reporter had 0 tests, so I also made sure to add these tests so we can catch these problems in the future. 
https://github.com/charlypoly/graphql-ruby-hive/blob/869c33fc4402b6e4e3c1df73c83b118cbe08ab4f/spec/graphql/graphql-hive/usage_reporter_spec.rb

cc: @charlypoly

---

`>= 2.3` might be less maintenance. We'd only have to update the pin if the next version breaks something again.

---

We'll only see this break in services that then update to 2.4 for example. But that might be worth taking the risk

---

I updated this! `'>= 2.3', '< 3'`

We do have services that use our graphql library but not hive, so this gives at least some freedom.